server/graphql/document/src/json_schema_loader.rs

```diff
@@ -0,0 +1,25 @@
+use repository::{JSONSchema, JsonSchemaRepository, RepositoryError, StorageConnectionManager};
+
+use async_graphql::dataloader::*;
+use async_graphql::*;
+use std::collections::HashMap;
+
+pub struct JsonSchemaLoader {
+    pub connection_manager: StorageConnectionManager,
+}
+
+#[async_trait::async_trait]
+impl Loader<String> for JsonSchemaLoader {
+    type Value = JSONSchema;
+    type Error = RepositoryError;
+
+    async fn load(&self, ids: &[String]) -> Result<HashMap<String, Self::Value>, Self::Error> {
+        let connection = self.connection_manager.connection()?;
```

Comment:
We've transitioned to using services within loaders. Although this looks pretty clean.

Reply:
I added an issue for this kind of clean up tasks:
https://github.com/openmsupply/open-msupply/issues/74

Reply:
thanks